docs/components/plugins/api/APISectionUtils.tsx

```diff
@@ -174,12 +174,10 @@ const hardcodedTypeLinks: Record<string, string> = {
     'https://www.typescriptlang.org/docs/handbook/utility-types.html#excludeuniontype-excludedmembers',
   ExpoConfig:
     'https://github.com/expo/expo/blob/main/packages/%40expo/config-types/src/ExpoConfig.ts',
-  ExpoUpdatesManifest: '/versions/latest/sdk/updates/#expoupdatesmanifest',
   File: 'https://developer.mozilla.org/en-US/docs/Web/API/File',
   FileList: 'https://developer.mozilla.org/en-US/docs/Web/API/FileList',
   IterableIterator:
     'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator',
-  Manifest: '/versions/latest/sdk/constants/#manifest',
```

Comment:
`Manifest` is a type reused by different packages, so we should retain it in the hardcoded links, but updated the URL to point to the new reference page. Otherwise type will not be linked, or link will not work on other API reference pages.

Reply:
@wschurman I just added the preview tag to look at the preview, and it looks like you need to annotate the `Manifest` definition in `Updates.types.ts` to have `NewManifest` and `BareManifest` point to the right types in `expo-manifest`.

Reply:
https://github.com/expo/expo/pull/27674 addresses this for 50 and unversioned (51). But leaves 48 and 49 in a broken state. I think this tradeoff is ok.

There are a couple issues with having it be hardcoded:
- it is hardcoded to latest, which is incorrect for most sdk versions since it changes
- the hardcoded link is not technically correct for the most recent SDK versions (the links were inverted so now expo-manifests is the source of truth). The links to expo-manifest also didn't work.
- the hardcoded link was not technically correct for the old SDK versions either. It could be a modern manifest or embedded manifest, thus the need for the disambiguation. I think the issue is that it does `typeof` and the type system doesn't work for this? https://github.com/expo/expo/blob/sdk-49/packages/expo-updates/src/Updates.types.ts#L34

Reply:
Thanks for thinking about that! The reason for hard code was that we have some problem with handling manifest in the past, when it was a part of `constants` package and those are basically workarounds to provide at least some information for the users. In general we should not internal links in hardcoded URL array.

There are also a limitation on the TypeDoc site which forces us the write our code and typing in certain way, to make sure that they are extracted correctly, and in the past changing the manifest code in a way we wanted to was not possible.

From the changes history, this hardcode was added quite recently in - https://github.com/expo/expo/pull/21237 - and looking at changes in this changeset we might don't need that hardcode for `Manifest` anymore. Instead maybe we should handle `ExpoUpdatesManifest` and `EmbeddedManifest` differently since they are imported from the other package, or we can just re-export those types from the `expo-updates` package (but this might cause also few more missing types, if we nest objects).

Reply:
>  Instead maybe we should handle ExpoUpdatesManifest and EmbeddedManifest differently since they are imported from the other package, or we can just re-export those types from the expo-updates package (but this might cause also few more missing types, if we nest objects).

Yep, https://github.com/expo/expo/pull/27674 does precisely this!

Reply:
The issue with "we might don't need that hardcode for Manifest anymore" is that that only applies to the most recent SDK versions, whereas the hardcoding applies to all SDK versions. So we've reached a logical impasse: it must exist for 48 and 49 (but be versioned, not point to latest), and is incorrect to exist for 50 and 51+.

So the solution I went with was to fix 50 and 50+ using that PR and then leave 48 and 49 in a broken state. I may be able to do some treatment in a follow-up where the hardcoded links are sdk-version-specific, but that seemed like more trouble than it was worth since we really shouldn't have internal links in the hardcoded URL array anyways and it would only be necessary for that. Also I don't think we easily have access to the current SDK version in the link rendering code.

Reply:
I added a preview tag to that PR as well (which is on the top of this stack) so one can see the final result that I describe.